docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc

```diff
@@ -0,0 +1,207 @@
+
+**Solution:** On the **Review and configure for deployment** page, make sure **Runtime** is set to **Dockerfile**, then verify that you saved your Dockerfile in one of the expected directories.
+
+[TIP]
+====
+Test your Dockerfile in a local environment first to make sure it's valid.
+====
+
+[#repository requirements]
+* **Your repository is missing a Dockerfile or devfile.**
++
+If your code import fails because your repository is missing either a Dockerfile or devfile, check one of the following readme files for guidance, depending on the runtime you want to build and deploy with. These readme files provide details like required ports, file locations, commands, and more. 
++
+**Solutions**
++
+** https://github.com/devfile-samples/devfile-sample-go-basic.git[Go]
+** https://github.com/nodeshift-starters/devfile-sample.git[Node.js]
+** https://github.com/devfile-samples/devfile-sample-python-basic.git[Python]
+** https://github.com/devfile-samples/devfile-sample-code-with-quarkus.git[Quarkus]
+** https://github.com/devfile-samples/devfile-sample-java-springboot-basic.git[Spring Boot]
+
+* **Your repository contains a devfile, a Dockerfile, and a Kubernetes YAML file.**
++
+{ProductName} looks for a devfile to get custom instructions for building and deploying your application. If your devfile references both a Dockerfile and a Kubernetes YAML file, your code import could fail if those two resources are not accessible at the file locations your devfile points to.
++
+**Solutions** 
++
+. Make sure that your `devfile.yaml` points to the correct directories for both your Dockerfile and your Kubernetes YAML file. 
+. Make sure that neither of these resources is in a private repository that requires access authentication.
+. Make sure that your devfile is in one of these expected locations:
++
+[#expected devfile locations] 
+.Expected devfile locations
+** devfile.yaml
+** .devfile.yaml
+** .devfile or devfile.yaml
+** .devfile or .devfile.yaml
+
+* **Your repository does not fit within the predefined {ProductName} supported runtime list.**
++
+Some repositories require custom build or deployment instructions; for example, a deployment that includes a Kubernetes resource other than a YAML file.
++
+**Solutions**
++
+[#custom build and deployment instructions]
+** **Configure a custom build.** Include a Dockerfile that can build your application, then save it to one of the <<expected Dockerfile locations>>. **Note:** If your build is custom but your deployment is not, you do not have to include a `devfile.yaml` file in your repository.
+** **Configure a custom deployment.** Include the following files in your repository to provide {ProductName} with custom deployment instructions:
+*** A standard Kubernetes YAML file
+*** A custom-build Dockerfile
+*** A devfile that points to your Kubernetes YAML file and Dockerfile. **Note:** Make sure that your devfile is in one of the <<expected devfile locations>>.
```

Comment:
We should also add a link to https://devfile.io/docs/2.2.0/innerloop-vs-outerloop and tell the user to refer to the outerloop section of the doc to understand the requirements on the content of the devfile.